server/sonar-server/src/test/java/org/sonar/server/computation/measure/MeasureAssert.java

```diff
@@ -0,0 +1,333 @@
+          Measure.ValueType.DOUBLE, actual.getValueType());
+    }
+
+    return this;
+  }
+
+  public MeasureAssert hasData(String expected) {
+    isNotNull();
+
+    if (!Objects.equals(actual.getData(), expected)) {
+      failWithMessage("Expected data of Measure to be <%s> but was <%s>", expected, actual.getData());
+    }
+
+    return this;
+  }
+
+  public MeasureAssert hasNoData() {
+    isNotNull();
+
+    if (actual.getData() == null) {
+      failWithMessage("Expected Measure to have no data but was <%s>", actual.getData());
+    }
+
+    return this;
+  }
+
+  public MeasureAssert hasDescription(String expected) {
+    isNotNull();
+
+    if (!Objects.equals(actual.getDescription(), expected)) {
+      failWithMessage("Expected description of Measure to be <%s> but was <%s>", expected, actual.getDescription());
+    }
+
+    return this;
+  }
+
+  public MeasureAssert hasNoDescription() {
+    isNotNull();
+
+    if (actual.getDescription() != null) {
+      failWithMessage("Expected Measure to have no description but it was <%s>", actual.getDescription());
+    }
+
+    return this;
+  }
+
+  public MeasureAssert hasNoQualityGateStatus() {
+    isNotNull();
+    hasQualityGateStatus();
+
+    if (actual.hasQualityGateStatus()) {
+      failWithMessage("Expected Measure to have no QualityGateStatus but it did");
+    }
+
+    return this;
+  }
+
+  public MeasureAssert hasQualityGateLevel(Measure.Level expected) {
+    isNotNull();
+    hasQualityGateStatus();
+
+    if (actual.getQualityGateStatus().getStatus() != expected) {
+      failWithMessage("Expected Level of QualityGateStatus of Measure to be <%s> but was <%s>", expected, actual.getQualityGateStatus().getStatus());
+    }
+
+    return this;
+  }
+
+  public MeasureAssert hasQualityGateText(String expected) {
+    isNotNull();
+    hasQualityGateStatus();
+
+    if (!Objects.equals(actual.getQualityGateStatus().getText(), expected)) {
+      failWithMessage("Expected text of QualityGateStatus of Measure to be \n<%s>\n but was \n<%s>", expected, actual.getQualityGateStatus().getText());
+    }
+
+    return this;
+  }
+
+  public MeasureAssert hasNoQualityGateText() {
+    isNotNull();
+    hasQualityGateStatus();
+
+    if (actual.getQualityGateStatus().getText() == null) {
+      failWithMessage("Expected Measure to have no QualityGate Test but was <%s>", actual.getQualityGateStatus().getText());
+    }
+
+    return this;
+  }
+
+  private void hasQualityGateStatus() {
+    if (!actual.hasQualityGateStatus()) {
+      failWithMessage("Expected Measure to have a QualityGateStatus but it did not");
+    }
+  }
+
+  public MeasureAssert hasNoVariations() {
+    isNotNull();
+
+    if (actual.hasVariations()) {
+      failWithMessage("Expected Measure to have a no Variations but it did");
+    }
+
+    return this;
+  }
+
+  public MeasureAssert hasVariation1(double expected) {
```

Comment:
`hasVariation1(1d)` is shorter and much clearer than `hasVariation(1d, 1)`
first version does not require to know what the second parameter is and there is no wondering which parameter is to be asserted